doc/source/overview.rst

```diff
@@ -117,6 +117,30 @@ Institutional Partners
 
 The information about current institutional partners can be found on `pandas website page <https://pandas.pydata.org/about.html>`__.
 
+Modules Privacy
+----------------
+
+The following modules in pandas are considered PRIVATE:
```

Comment:
Two things

1. A bit of context would be helpful
2. I worry about this going out of date

So how about something like

    In addition to sub-packages and modules prefixed with `_`, the following sub-packages and modules are considered private and may change or functionality without warning.
    
    * `pandas.compat`
    * `pandas.core` and all submodules
    * `pandas.util` and all submodules

    While pandas developers have always considered everything in `pandas.core` "private", we deemed simply renaming it to `pandas._core` too disruptive.

    Pandas' public API consists of everything listed in :ref:`api`.

Alternatively, we can list out modules

Reply:
Agreed.